utility_analysis_new/tests/contribution_bounders_test.py

```diff
@@ -0,0 +1,95 @@
+        params = CrossAndPerPartitionContributionParams(
+            max_partitions_contributed, max_contributions_per_partition)
+
+        bounder = contribution_bounders.SamplingCrossAndPerPartitionContributionBounder(
+        )
+        return list(
+            bounder.bound_contributions(input, params,
+                                        pipeline_dp.LocalBackend(),
+                                        _create_report_generator(),
+                                        aggregate_fn))
+
+    def test_contribution_bounding_empty_col(self):
+        input = []
+        max_partitions_contributed = max_contributions_per_partition = 2
+        bound_result = self._run_contribution_bounding(
+            input, max_partitions_contributed, max_contributions_per_partition)
+
+        self.assertEmpty(bound_result)
+
+    def test_contribution_bounding_bound_input_nothing_dropped(self):
+        input = [("pid1", 'pk1', 1), ("pid1", 'pk1', 2), ("pid1", 'pk2', 3),
+                 ("pid1", 'pk2', 4)]
+        max_partitions_contributed = max_contributions_per_partition = 2
+        bound_result = self._run_contribution_bounding(
+            input, max_partitions_contributed, max_contributions_per_partition)
+
+        expected_result = [(('pid1', 'pk2'), 2), (('pid1', 'pk1'), 2)]
+        self.assertEqual(set(expected_result), set(bound_result))
+
+    def test_contribution_bounding_per_partition_bounding_not_applied(self):
+        input = [("pid1", 'pk1', 1), ("pid1", 'pk1', 2), ("pid1", 'pk2', 3),
+                 ("pid1", 'pk2', 4), ("pid1", 'pk2', 5), ("pid2", 'pk2', 6)]
+        max_partitions_contributed, max_contributions_per_partition = 5, 2
+        bound_result = self._run_contribution_bounding(
+            input, max_partitions_contributed, max_contributions_per_partition)
+
+        expected_result = [(('pid1', 'pk1'), 2), (('pid1', 'pk2'), 3),
+                           (('pid2', 'pk2'), 1)]
+        # Check per-partition contribution limits are not enforced.
+        self.assertEqual(set(expected_result), set(bound_result))
+
+    def test_contribution_bounding_cross_partition_bounding_not_applied(self):
+        input = [("pid1", 'pk1', 1), ("pid1", 'pk1', 2), ("pid1", 'pk2', 3),
+                 ("pid1", 'pk2', 4), ("pid1", 'pk2', 5), ("pid1", 'pk3', 6),
+                 ("pid1", 'pk4', 7), ("pid2", 'pk4', 8)]
+        max_partitions_contributed = 3
+        max_contributions_per_partition = 5
+
+        bound_result = self._run_contribution_bounding(
+            input, max_partitions_contributed, max_contributions_per_partition)
+
+        expected_result = [(('pid1', 'pk1'), 2), (('pid1', 'pk2'), 3),
+                           (('pid1', 'pk3'), 1), (('pid1', 'pk4'), 1),
+                           (('pid2', 'pk4'), 1)]
+        # Check per- and cross-partition contribution limits are not enforced.
+        self.assertEqual(set(expected_result), set(bound_result))
```

Comment:
Please add 

```
if __name__ == '__main__':
    absltest.main()
```

[example](https://github.com/OpenMined/PipelineDP/blob/dee896300bcf67b48363a41170e9acfefe94c77d/tests/combiners_test.py#L529)

Otherwise tests wouldn't run.